### PR TITLE
(Minor) More tidying for non host alignment functions

### PR DESCRIPTION
**Pull request by jshoe**
- Small changes for non host alignment functions
- Test run here: https://staging.idseq.net/samples/587

### CODE REVIEW

File: idseq_pipeline/commands/non_host_alignment_functions.py [hunk 1]
```diff
@@ -158,45 +167,54 @@ def subsample_helper(input_file, records_to_keep, type_, output_file):
...
-            missing = records_to_keep - kept_read_ids
-            examples = sorted(random.sample(missing, min(10, len(missing))))
-            assert kept_read_ids == records_to_keep, "Not all desired read IDs were found in the file: {}\nMissing: {}".format(
-                input_file, examples)
+
+    # Error checking for mismatch
+    if type_ == "record_indices" and len(kept_read_ids) != len(records_to_keep):
+        msg = "WARNING: Repeated read IDs in {input_file}: Found {kri} read " \
+              "IDs in {rtk} sampled rows.".format(
+        input_file=input_file,
```
Comment (jameshowardwang) [inline idseq_pipeline/commands/non_host_alignment_functions.py]: are these indented properly?

File: idseq_pipeline/commands/non_host_alignment_functions.py [hunk 2]
```diff
@@ -353,80 +380,104 @@ def get_map(m8_file):
...
-                    }
-                    aggregation[agg_key] = agg_bucket
-                agg_bucket['count'] += 1
-                agg_bucket['sum_percent_identity'] += percent_identity
-                agg_bucket['sum_alignment_length'] += alignment_length
-                agg_bucket['sum_e_value'] += e_value
-                # Chomp off the lowest rank as we aggregate up the tree
-                agg_key = agg_key[1:]
+        if any_hits_to_remove(cleaned_hit_taxids_all_levels):
+            hit_line = hit_f.readline()
```
Comment (jameshowardwang) [inline idseq_pipeline/commands/non_host_alignment_functions.py]: not sure about this change. now there's a bit of code dupe...

File: idseq_pipeline/commands/non_host_alignment_functions.py (hunk 1 above)
Comment (jshoe) [reply idseq_pipeline/commands/non_host_alignment_functions.py]: fixed

File: idseq_pipeline/commands/non_host_alignment_functions.py (hunk 2 above)
Comment (jshoe) [reply idseq_pipeline/commands/non_host_alignment_functions.py]: changed back

**Pull request merged**